pkg/controller/container-runtime-config/helpers.go

```diff
@@ -954,3 +955,108 @@ func generateSigstoreRegistriesdConfig(clusterScopePolicies map[string]signature
 	}
 	return data, nil
 }
+
+func addScopeMirrorsSigstoreRegistriesdConfig(registriesDockerConfig map[string]dockerConfig, scope string, icspRules []*apioperatorsv1alpha1.ImageContentSourcePolicy, idmsRules []*apicfgv1.ImageDigestMirrorSet, itmsRules []*apicfgv1.ImageTagMirrorSet, sigstoreAttachment dockerConfig) {
+	for _, icsp := range icspRules {
+		scopeIsSource := false
+		for _, rdm := range icsp.Spec.RepositoryDigestMirrors {
+			if strings.HasPrefix(rdm.Source, scope) {
+				for _, mirror := range rdm.Mirrors {
+					registriesDockerConfig[mirror] = sigstoreAttachment
+				}
+				if rdm.Source == scope {
+					scopeIsSource = true
+				}
+			}
+		}
+		if scopeIsSource {
+			continue
+		}
+
+		maxSourceLen := 0
+		var mirrors []string
+		for _, rdm := range icsp.Spec.RepositoryDigestMirrors {
+			if strings.HasPrefix(scope, rdm.Source) {
+				if len(rdm.Source) > maxSourceLen {
+					maxSourceLen = len(rdm.Source)
+					mirrors = rdm.Mirrors
+				}
+			}
+		}
+		if maxSourceLen > 0 {
+			for _, mirror := range mirrors {
+				registriesDockerConfig[mirror] = sigstoreAttachment
+			}
+		}
+	}
```

Comment:
@mtrmac What do you think about the above loop for adding the relevant mirrors to the Sigstore attachment configuration file? I am not sure if we should just do:
```
for source in icsp/idms/itms {
    if hasPrefix(source, scope) || hasPrefix(scope, source) {
        addMirrorsToFile()
    }
}
```